pkg/controller/servicebindingrequest/planner.go

```diff
@@ -88,31 +89,46 @@ func (p *Planner) Plan() (*Plan, error) {
 	relatedResources := make([]*RelatedResource, 0)
 	for _, s := range selectors {
 
+		if s.Namespace == nil {
+			s.Namespace = &ns
+		}
+
 		bssGVK := schema.GroupVersionKind{Kind: s.Kind, Version: s.Version, Group: s.Group}
 
+		// Start with looking up if the resource exists
+		// If yes, errors during lookups of the CRD and
+		// the CRD could be ignored.
+		cr, err := p.searchCR(s)
+		if err != nil {
+			return nil, err
+		}
+
 		// resolve the CRD using the service's GVK
 		crd, err := p.searchCRD(bssGVK)
 		if err != nil {
-			return nil, err
+			// expected this to work, but didn't
+			// if k8sError.IsNotFound(err) {...}
+			p.logger.Error(err, "Probably not a CRD")
```

Comment:
Actually no, I'm intentionally letting the control to flow into the next lines of code. if the CRD lookup failed, I would consider this to not be a CRD. The backing service is probably a 'general' kubernetes object.

Reply:
In this case, perhaps this branch should be extended until right before calling `buildCRDDescriptionFromCR`, since everything that happens in between is setting up the CRDDescription for that call and will not matter if `crd` is nil.

Reply:
You mean, if the CRD is absent, it is pointless to lookup the CSV too, is it?

Reply:
Done https://github.com/redhat-developer/service-binding-operator/pull/367/commits/ae26e71743cd2b496ff23e96d1d4d25b9ecfcf8c

```
if crd not found {
   log("probably not a crd")
}else{
    // continue looking for csv metadata
}

// lookup CR annotations

```